package packExceptions;
/* This exception is thrown when an attempt of collection access fails because
the position we tried to access does not belongs to the indexes range */
public class AccessOutOfBoundariesException extends Exception{
	public AccessOutOfBoundariesException() {
		super();
	}
	public AccessOutOfBoundariesException(String message) {
		super(message);
	}
}
/*This exception is thrown when we want to perform some illegal operation
on a empty collection */
public class EmptyCollectionException extends Exception{
	public EmptyCollectionException() {
		super();
	}
	public EmptyCollectionException(String message) {
		super(message);
	}
}


package packUser;

public class UnregisteredIDException extends Exception{
	public UnregisteredIDException() {
		super("No person registered on the social network with this ID has been found.");
	}
	
	@Override
	public String getMessage() {
		return super.getMessage();
	}

}


/* The following exceptions are 
thrown in case the atributes of any object of the class Person is created 
is considered incorrect */

package packExceptions;
public class ExistingID extends Exception {
	public ExistingID() {
		super("Already existing ID. Please change it.");
	}
	@Override
	public String getMessage() {
		return super.getMessage();
	}
}
package packExceptions;
public class EmptyID extends Exception{
	public EmptyID() {
		super("ID is compulsory.");
	}
	@Override
	public String getMessage() {
		return super.getMessage();
	}
}



